### PR TITLE
Fix parsing of Django `path` patterns

### PR DESCRIPTION
**Pull request by sentrivana**
TL;DR: Our way of [parsing](https://github.com/getsentry/sentry-python/blob/085595b5f02931a3268c2de2a58b6986f3766d75/sentry_sdk/integrations/django/transactions.py#L48) named group regexes out of Django URL patterns is [not ideal](https://github.com/getsentry/sentry-python/issues/2392#issuecomment-1755797139). If someone is using (Django 2.0+) `<converter:parameter>` `path` patterns, we can parse those directly without turning them into regexes first and potentially hitting our limitations there.

### Background

When instrumenting Django for performance monitoring, the Sentry Python SDK will by default name the transactions based on the URL pattern rule that was used to serve the request.

There are two main ways of creating a URL pattern in Django: by using regexes in the rule directly, or (since Django 2.0) by using the custom `<converter:parameter>` syntax, like so:

```python
urlpatterns = [
    path('articles/<int:year>/<int:month>/', views.month_archive),
]
```

These `<converter:parameter>` style patterns are also regexes in the background. Each of the converters defines its own regex (see e.g. the [IntConverter](https://github.com/django/django/blob/7bb741d787ba360a9f0d490db92e22e0d28204ed/django/urls/converters.py#L5-L12)) that is then used for matching URLs to the patterns.

The SDK is [parsing](https://github.com/getsentry/sentry-python/blob/085595b5f02931a3268c2de2a58b6986f3766d75/sentry_sdk/integrations/django/transactions.py#L48) both types of patterns with the aim of replacing all named groups with placeholders, so that e.g. the URL pattern above creates transactions with the name `articles/{year}/{month}`. To do this, the URL pattern is first turned into its regex form and then parsed with the RavenResolver.

This is not a problem with the default set of converters Django offers, but if folks define [their own custom converters](https://github.com/getsentry/sentry-python/issues/2446) whose regexes don't play nice with our RavenResolver, we fail to extract the transaction name correctly. This can be prevented by parsing the original, simpler URL pattern instead without turning it into its regex form.

Note that this doesn't solve all transaction name problems as `re_path`s are still vulnerable, but it should make new-style `path`s work.

Closes https://github.com/getsentry/sentry-python/issues/2446

### CODE REVIEW

File: sentry_sdk/integrations/django/transactions.py [hunk 1]
```diff
@@ -56,11 +63,23 @@ def _simplify(self, pattern):
         To:
         > "{sport_slug}/athletes/{athlete_slug}/"
         """
+        # "new-style" path patterns can be parsed directly without turning them
+        # into regexes first
+        if DJANGO_VERSION >= (2, 0):
```
Comment (sl0thentr0py) [inline sentry_sdk/integrations/django/transactions.py:68]: can we move this check and import to the top and make this branch `if RoutePattern is not None:`

**Review by sl0thentr0py — APPROVED** [review]
nice work! don't really know how the resolver works in detail but looks ok high level
